src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,254 @@
+import React, { cloneElement, useCallback, useEffect, useMemo } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import styles from "./Modal.module.scss";
+import { useA11yDialog } from "./a11yDialog";
+import { ModalContent, ModalFooter, ModalHeader } from "components";
+import { clearAllBodyScrollLocks, disableBodyScroll, enableBodyScroll } from "body-scroll-lock";
+import useAnimationProps from "components/Modal/useAnimationProps";
+import { useKeyEvent } from "../../hooks";
+
+export const MODAL_WIDTH = {
+  DEFAULT: "default",
+  FULL_WIDTH: "full_width"
+};
+
+const isModalHeader = child => child.type === ModalHeader;
+const isModalContent = child => child.type === ModalContent;
+const isModalFooter = child => child.type === ModalFooter;
+
+const Modal = ({
+  classNames,
+  id,
+  show,
+  title,
+  description,
+  onClose,
+  isAlertDialog,
+  children,
+  triggerElement,
+  width,
+  hideCloseButton,
+  closeButtonAriaLabel
+}) => {
+  const [instance, attr] = useA11yDialog({
+    id,
+    isAlertDialog
+  });
+
+  useKeyEvent({
+    callback: event => {
+      if (instance?.$el.contains(document.activeElement)) {
+        event.stopPropagation();
+        closeDialogIfNeeded();
+      }
+    },
+    capture: true,
+    keys: ["Escape"]
+  });
+
+  const closeDialogIfNeeded = useCallback(() => {
+    if (!isAlertDialog) {
+      onClose();
+    }
+  }, [isAlertDialog, onClose]);
+
+  const getAnimationProps = useAnimationProps(triggerElement, instance);
+
+  // clear all scroll locks on unmount (just to be safe)
+  useEffect(() => {
+    return () => clearAllBodyScrollLocks();
+  }, []);
+
+  // lock body on modal show
+  useEffect(() => {
+    instance?.on("show", () => disableBodyScroll(instance.$el, { reserveScrollBarGap: true }));
+    instance?.on("hide", () => enableBodyScroll(instance.$el));
+    return () => {
+      instance?.off("show");
+      instance?.off("hide");
+    };
+  }, [instance]);
+
+  useEffect(() => {
+    instance?.on("hide", () => onClose());
+    return () => {
+      instance?.off("hide");
+    };
+  }, [instance]);
+
+  // show/hide and animate the modal
+  useEffect(() => {
+    if (instance) {
+      const animate = instance?.$el.childNodes[1].animate;
+      if (show) {
+        instance?.show();
+        if (animate) instance?.$el.childNodes[1].animate?.(...getAnimationProps());
+      } else {
+        if (animate) {
+          const animation = instance?.$el.childNodes[1]?.animate(...getAnimationProps(true));
+          animation?.finished.then(() => instance?.hide());
+        } else {
+          instance?.hide();
+        }
+      }
+    }
+  }, [show, instance, getAnimationProps]);
```

Comment:
I think you can improve readability a bit with the same functionality, WDYT?

```suggestion
  if(!instance) {
    return;
  }
  const animate = instance.$el.childNodes[1].animate;
  if (show) {
    instance?.show();
    animate?.(...getAnimationProps());
  } else if (animate) {
    const animation = animate(...getAnimationProps(true));
    animation?.finished.then(() => instance?.hide());
  } else {
    instance.hide();
  }
  }, [show, instance, getAnimationProps]);
```

Reply:
done